asn1ate/asn2quickder.py

```diff
@@ -0,0 +1,321 @@
+		}
+		ik.pack_funmap = {
+			DefinedType: ik.packDefinedType,
+			ValueAssignment: ik.ignore_node,
+			TypeAssignment: ik.packTypeAssignment,
+			TaggedType: ik.packTaggedType,
+			SimpleType: ik.packSimpleType,
+			BitStringType: ik.packSimpleType,
+			ValueListType: ik.ovlySimpleType,
+			SequenceType: ik.packSequenceType,
+			SetType: ik.packSetType,
+			ChoiceType: ik.packChoiceType,
+			SequenceOfType: ik.packSequenceOfType,
+			SetOfType: ik.packSetOfType,
+			ComponentType: ik.packSimpleType,  #TODO#
+		}
+
+	def newcomma (ik, comma, firstcomma=''):
+		ik.comma0 = firstcomma
+		ik.comma1 = comma
+
+	def comma (ik):
+		ik.wout (ik.comma0)
+		ik.comma0 = ik.comma1
+
+	def close (ik):
+		ik.outfile.close ()
+
+	def generate_head (ik):
+		ik.wout ('/*\n * asn2quickder output for ' + ik.semamod.name + ' -- automatically generated\n *\n * For information on Quick `n\' Easy DER, see https://github.com/vanrein/quick-der\n *\n * For information on the code generator, see https://github.com/vanrein/asn2quickder\n *\n */\n\n\n#include <quick-der/api.h>\n\n\n')
+		# ik.wout ('/* This module ' + toCsym (ik.semamod.name) + ' depends on:\n')
+		# for rm in ik.refmods:
+		# 	ik.wout (' *   ' + toCsym (rm.name) + '\n')
+		# if len (ik.refmods) == 0:
+		# 	ik.wout (' *   (no other modules)\n')
+		# ik.wout (' */\n\n')
+		closer = ''
+		for rm in ik.semamod.imports.module2symbols.keys ():
+			rmfn = toCsym (rm.rsplit ('.', 1) [0]).lower ()
+			ik.wout ('#include <quick-der/' + rmfn + '.h>\n')
+			closer = '\n\n'
+		ik.wout (closer)
+		closer = ''
+		for rm in ik.semamod.imports.module2symbols.keys ():
+			rmfn = toCsym (rm.rsplit ('.', 1) [0]).lower ()
+			for sym in ik.semamod.imports.module2symbols [rm]:
+				ik.wout ('typedef DER_OVLY_' + toCsym (rmfn) + '_' + toCsym (sym) + ' DER_OVLY_' + toCsym (ik.unit) + '_' + toCsym (sym) + ';\n')
+				closer = '\n\n'
+		ik.wout (closer)
+		closer = ''
+		for rm in ik.semamod.imports.module2symbols.keys ():
+			rmfn = toCsym (rm.rsplit ('.', 1) [0]).lower ()
+			for sym in ik.semamod.imports.module2symbols [rm]:
+				ik.wout ('#define DER_PACK_' + toCsym (ik.unit) + '_' + toCsym (sym) + ' DER_PACK_' + toCsym (rmfn) + '_' + toCsym (sym) + '\n')
+				closer = '\n\n'
+		ik.wout (closer)
+
+	def generate_tail (ik):
+		ik.wout ('\n\n/* asn2quickder output for ' + ik.semamod.name + ' ends here */\n')
+
+	def generate_ovly (ik):
+		ik.wout ('\n\n/* Overlay structures with ASN.1 derived nesting and labelling */\n\n')
+		for assigncompos in dependency_sort (ik.semamod.assignments):
+			for assign in assigncompos:
+				ik.generate_ovly_node (assign)
+
+	def generate_pack (ik):
+		ik.wout ('\n\n/* Parser definitions in terms of ASN.1 derived bytecode instructions */\n\n')
+		for assigncompos in dependency_sort (ik.semamod.assignments):
+			for assign in assigncompos:
+				tnm = type (assign)
+				if tnm in ik.pack_funmap:
+					ik.pack_funmap [tnm] (assign)
+				else:
+					print 'No pack generator for ' + str (tnm)
```

Comment:
I use Python 3 and support both Py3k and Python 2.x. Can you use `print` with parentheses to maintain this?